osu.Game/Screens/Play/Player.cs

```diff
@@ -1171,6 +1174,10 @@ public override void OnSuspending(ScreenTransitionEvent e)
         {
             screenSuspension?.RemoveAndDisposeImmediately();
 
+            // If these are not disposed, audio volume dimming can get stuck.
+            retryOverlay?.RemoveAndDisposeImmediately();
+            exitOverlay?.RemoveAndDisposeImmediately();
```

Comment:
Do you think we should maybe add a

```diff
diff --git a/osu.Game/Screens/Play/Player.cs b/osu.Game/Screens/Play/Player.cs
index 38e3fcd38d..9988bbcd93 100644
--- a/osu.Game/Screens/Play/Player.cs
+++ b/osu.Game/Screens/Play/Player.cs
@@ -1172,6 +1172,8 @@ protected virtual void StartGameplay()
 
         public override void OnSuspending(ScreenTransitionEvent e)
         {
+            Debug.Assert(!ValidForResume);
+
             screenSuspension?.RemoveAndDisposeImmediately();
 
             // If these are not disposed, audio volume dimming can get stuck.

```

here for safety? I *think* it should be fine but player code is twisty enough to have me not wholly confident that that's the case.

Reply:
Sure, added.